codec-classes-quic/src/main/java/io/netty/incubator/codec/quic/QuicheQuicSslContext.java

```diff
@@ -95,17 +96,28 @@
         } else {
             keyManager = chooseKeyManager(keyManagerFactory);
         }
+        sessionCache = server ? null : new QuicClientSessionCache();
         int verifyMode = server ? boringSSLVerifyModeForServer(this.clientAuth) : BoringSSL.SSL_VERIFY_PEER;
         nativeSslContext = new NativeSslContext(BoringSSL.SSLContext_new(server, applicationProtocols,
                 new BoringSSLHandshakeCompleteCallback(engineMap),
                 new BoringSSLCertificateCallback(engineMap, keyManager, password),
                 new BoringSSLCertificateVerifyCallback(engineMap, trustManager),
                 mapping == null ? null : new BoringSSLTlsextServernameCallback(engineMap, mapping),
-                keylog ? new BoringSSLKeylogCallback() : null, verifyMode,
+                keylog ? new BoringSSLKeylogCallback() : null,
+                server ? null : new BoringSSLSessionCallback(engineMap, sessionCache), verifyMode,
                 BoringSSL.subjectNames(trustManager.getAcceptedIssuers())));
         apn = new QuicheQuicApplicationProtocolNegotiator(applicationProtocols);
-        this.sessionCacheSize = BoringSSL.SSLContext_setSessionCacheSize(nativeSslContext.address(), sessionCacheSize);
-        this.sessionTimeout = BoringSSL.SSLContext_setSessionCacheTimeout(nativeSslContext.address(), sessionTimeout);
+        if (this.sessionCache != null) {
```

Comment:
Should we reuse?
setSessionTimeout((int)sessionTimeout);
setSessionCacheSize((int) sessionCacheSize);

Reply:
calling own non static methods in the constructor is not really something I like (due possible NPE etc) so lets keep it